packages/axe-core-api/lib/axe/core.rb

```diff
@@ -40,14 +40,18 @@ def use_run_partial
 
     def assert_frame_ready
       begin
-        ready = Timeout.timeout(10) {
-          @page.evaluate_script <<-JS
-            document.readyState === 'complete'
+        wait = Selenium::WebDriver::Wait.new(:timeout => 10)
+        wait.until { 
+          readyState = @page.evaluate_script <<-JS
+            document.readyState
```

Comment:
I know this has been here for a while, but have you tested if we even need this anymore? I know for Selenium they have implemented the WebDriver as per the [spec](https://w3c.github.io/webdriver/#navigate-to) which by default is set to so we _should_ be waiting for the page finish loading before trying to inject: 

> An [HTTP session](https://w3c.github.io/webdriver/#dfn-http-session) has an associated page loading strategy, which is one of the keywords from the [table of page load strategies](https://w3c.github.io/webdriver/#dfn-table-of-page-load-strategies). This is initially set to [normal](https://w3c.github.io/webdriver/#dfn-normal-page-loading-strategy).

Reply:
If it's for backcompat with older selenium versions that we expect are missing that behavior, it'd be helpful to leave a comment explaining that

Reply:
I tested removing this against both latest 3 and 4, and I didn't find any issue against either the "fast" or "slow" URLs in my repro script. I'm removing it for this reason. If I'm misunderstanding, please let me know.
It seems odd that it would work on latest 3, but there may have been a fix in place since this was first written.